ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -419,18 +429,19 @@ private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStrea
    */
   private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, AdminRequest adminRequest)
       throws StoreException, IOException {
-    boolean hostHealthy = false; //used to determine if the host is ever unhealthy
+    ServerHealthStatus serverHealthStatus = ServerHealthStatus.BAD; //used to determine if the server is ever unhealthy
```

Comment:
this can be initiailiazed to GOOD and L437 can be removed.

Reply:
If the storageManager isn't a StorageManager then the healthstatus will return GOOD. So by checkng what storageManager is first, we know if we can check the replica/disk states otherwise we have no information about the server